### PR TITLE
jmh Blackhole

### PR DESCRIPTION
**Pull request by patriknw**

### CODE REVIEW

File: akka-bench-jmh/src/main/scala/akka/stream/GraphBuilderBenchmark.scala [hunk 1]
```diff
@@ -16,22 +17,22 @@ class GraphBuilderBenchmark {
   var complexity = 0
 
   @Benchmark
-  def flow_with_map(): Unit = {
-    MaterializationBenchmark.flowWithMapBuilder(complexity)
+  def flow_with_map(blackhole: Blackhole): Unit = {
+    blackhole.consume(MaterializationBenchmark.flowWithMapBuilder(complexity))
```
Comment (drewhk) [inline akka-bench-jmh/src/main/scala/akka/stream/GraphBuilderBenchmark.scala]: isn't it enough to expose the return type in the method? AFAIK that is equivalent to this (JMH should blackhole it automagically).

**Comment by ktoso** [comment]
Yup, as noted here:
https://github.com/akka/akka/commit/472d6c6266013bdc85c04cb6c54eca92e5b1e1c6

just return the value :)

-- 
Konrad `ktoso` Malawski
Akka <http://akka.io> @ Lightbend <http://lightbend.com>

On 9 March 2017 at 10:19:29, drewhk (notifications@github.com) wrote:

*@drewhk* commented on this pull request.
------------------------------

In akka-bench-jmh/src/main/scala/akka/stream/GraphBuilderBenchmark.scala
<https://github.com/akka/akka/pull/22510#discussion_r105117620>:

> @@ -16,22 +17,22 @@ class GraphBuilderBenchmark {
   var complexity = 0

   @Benchmark
-  def flow_with_map(): Unit = {
-    MaterializationBenchmark.flowWithMapBuilder(complexity)
+  def flow_with_map(blackhole: Blackhole): Unit = {
+    blackhole.consume(MaterializationBenchmark.flowWithMapBuilder(complexity))

isn't it enough to expose the return type in the method? AFAIK that is
equivalent to this (JMH should blackhole it automagically).

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/akka/akka/pull/22510#pullrequestreview-25980230>, or mute
the thread

**Comment by patriknw** [comment]
ok, feel free to fix, I'm in a call

**Comment by drewhk** [comment]
I don't mind it being explicitly blackholed, harder to miss.

**Comment by ktoso** [comment]
Yep I'll do that, working on the small tickets now then

**Pull request closed without merging**